sources/vega/snapshot/v1/snapshot.proto

```diff
@@ -4,77 +4,271 @@ package vega.snapshot.v1;
+  FORMAT_PROTO = 1;
+  // Compressed proto, might come in handy
+  FORMAT_PROTO_COMPRESSED = 2;
+  // Simple JSON serialisation, probably not going to be used
+  FORMAT_JSON = 3;
+}
+
+// Represents a set of time intervals that are used when querying for candle-stick data
 message Snapshot {
-  bytes hash = 1;
-  bytes state = 2;
+  uint64 height = 1;
+  Format format = 2;
+  uint32 chunks = 3;
+  bytes hash = 4;
+  bytes metadata = 5;
 }
 
-// Checkpoint aggregates the various engine snapshots
-message Checkpoint {
-  bytes governance = 1;
-  bytes assets = 2;
-  bytes collateral = 3;
-  bytes network_parameters = 4;
-  bytes delegation = 5;
-  bytes epoch = 6; // will just be an epoch event
-  bytes block = 7;
+message NodeHash {
+  string full_key = 1; // namespace<separator>key
+  string namespace = 2;
+  string key = 3;
+  string hash = 4;
+}
+
+message Metadata {
+  int64 version = 1;
+  repeated string chunk_hashes = 2;
+  repeated NodeHash node_hashes = 3; // hashes used by snapshot engine, different to chunk hash if data was split up, or combined into a single chunk
+}
+
+// Chunk is simply an as-is chunk belonging to a snapshot
+// not sure how usable this type would be
+message Chunk {
+  repeated Payload data = 1;
+}
+
+message Payload {
+  oneof data {
+    ActiveAssets active_assets = 1;
+    PendingAssets pending_assets = 2;
+    BankingWithdrawals banking_withdrawals = 3;
+    BankingDeposits banking_deposits = 4;
+    BankingSeen banking_seen = 5;
+    Checkpoint checkpoint = 6;
+    CollateralAccounts collateral_accounts = 7;
+    CollateralAssets collateral_assets = 8;
+    DelegationActive delegation_active = 9;
+    DelegationPending delegation_pending = 10;
+    GovernanceActive governance_active = 11;
+    GovernanceEnacted governance_enacted = 12;
+    StakingAccounts staking_accounts = 13;
+    MatchingBook matching_book = 14;
+    NetParams network_parameters = 15;
+    ExecutionMarkets execution_markets = 16;
+    MarketPositions market_positions = 17;
+    AppState app_state = 18;
+  }
+}
+
+message CollateralAccounts {
+  repeated vega.Account accounts = 1;
+}
+
+message CollateralAssets {
+  repeated vega.AssetDetails assets = 1;
+}
+
+message ActiveAssets {
+  repeated vega.AssetDetails assets = 1;
+}
+
+message PendingAssets {
+  repeated vega.AssetDetails assets = 1;
 }
 
-// AssetEntrty is a single (enabled) asset
-message AssetEntry {
+message Withdrawal {
+  string ref = 1;
+  vega.Withdrawal withdrawal = 2;
+}
+
+message Deposit {
   string id = 1;
-  vega.AssetDetails asset_details = 2;
+  vega.Deposit deposit = 2;
+}
+
+message TxRef {
+  string asset = 1;
+  uint64 block_nr = 2;
+  string hash = 3;
+  uint64 log_index = 4;
+}
+
+message BankingWithdrawals {
+  repeated Withdrawal withdrawals = 1;
+}
+
+message BankingDeposits {
+  repeated Deposit deposit = 1;
+}
+
+message BankingSeen {
+  repeated TxRef refs = 1;
+}
+
+message Checkpoint {
+  int64 next_cp = 1;
+}
+
+message DelegationActive {
```

Comment:
what about epochs?

Reply:
yes, still adding types, epoch will be added